### PR TITLE
Corrected get_by_label to use only labels in the ontology

### PR DESCRIPTION
**Pull request by francescalb**
# Description
get:by:label used to search wider than what ist was supposed to in the sense that is searched for prefLabels even when skos:prefLabel was not in the ontology.

Furthermore, documentation for get_by_label and get_by_label_all has been updated.

## Type of change

- [x] Bug fix.
- [ ] New feature.
- [x] Documentation update.
- [ ] Test update.

## Checklist

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

### CODE REVIEW

File: ontopy/ontology.py [hunk 1]
```diff
@@ -292,7 +305,7 @@ def get_by_label(
     ):
         """Returns entity with label annotation `label`.
 
-        Args:
+        Arguments:
            label: label so serach for.
```
Comment (francescalb) [inline ontopy/ontology.py:309]:
```suggestion
           label: label so search for.
```

File: ontopy/ontology.py [hunk 2]
```diff
@@ -396,25 +401,41 @@ def get_by_label(
...
+        label,
+        label_annotations=None,
+        prefix=None,
+        exact_match=False,
+    ):
         """Like get_by_label(), but returns a list with all matching labels.
 
-        Returns an empty list if no matches could be found.
+        Arguments:
+           label: label so serach for.
```
Comment (francescalb) [inline ontopy/ontology.py:421]:
```suggestion
           label: label so search for.
```

File: ontopy/ontology.py [hunk 3]
```diff
@@ -100,8 +100,9 @@ def get_ontology(
                 this Ontology subclass.  Defaults to `ontopy.Ontology`.
             label_annotations: Sequence of label IRIs used for accessing
                 entities in the ontology given that they are in the ontology.
-                Label IRIs not in the ontology will simply be ignored.
-                Detault to DEFAULT_LABEL_ANNOTATIONS module variable.
+                Label IRIs not in the ontology will need to be added to
+                ontologies in order to be accessible.
+                Default is None.
```
Comment (jesper-friis) [inline ontopy/ontology.py:105]: None (default), is equivalent to set `label_annotations` to the `DEFAULT_LABEL_ANNOTATIONS` module variable.

**Review by jesper-friis — APPROVED** [review]
Great work! Added a suggestion for a small change in docstring.

File: ontopy/ontology.py [hunk 4]
```diff
@@ -76,18 +76,33 @@ def __init__(self, *args, **kwargs):
...
+                  version of EMMO. Until first stable release
+                  emmo-inferred and emmo-development will be the same.
+            OntologyClass: If given and `base_iri` doesn't correspond
+                to an existing ontology, a new ontology is created of
+                this Ontology subclass.  Defaults to `ontopy.Ontology`.
+            label_annotations: Sequence of label IRIs used for accessing
+                entities in the ontology given that they are in the ontology.
+                Label IRIs not in the ontology will need to be added to
+                ontologies in order to be accessible.
+                Default is None.
```
Comment (francescalb) [inline ontopy/ontology.py:105]:
```suggestion
                ontologies in order to be accessible.
                Defaults to DEFAULT_LABEL_ANNOTATIONS if set to None.
```

**Pull request merged**